File: 506.cpp
#include <algorithm>

struct Solution {
    vector<string> findRelativeRanks(vector<int>& score) {
        vector<string> res(score.size());
        vector<int> rk(score.size());
        iota(rk.begin(), rk.end(), 0);
        sort(rk.begin(), rk.end(), [&score](int i, int j){return score[i] > score[j];});
        for (int idx = 0; idx < rk.size(); ++idx){
            int val = rk[idx];
            if (idx >= 3){
                res[val] = to_string(idx + 1);
            } else if (idx == 2) {
                res[val] = "Bronze Medal";
            } else if (idx == 1) {
                res[val] = "Silver Medal";
            } else {
                res[val] = "Gold Medal";
            }
        }
        return res;
    }
};

File: sol/506.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        ranks = sorted(range(len(score)), reverse=True, key=lambda i: score[i])
        for r, idx in enumerate(ranks):
            if r >= 3:
                score[idx] = str(r + 1)
            elif r == 2:
                score[idx] = "Bronze Medal"
            elif r == 1:
                score[idx] = "Silver Medal"
            else:
                score[idx] = "Gold Medal"
        return score
